mypyc/genops.py

```diff
@@ -1205,13 +1206,58 @@ def py_get_attr(self, obj: Value, attr: str, line: int) -> Value:
         key = self.load_static_unicode(attr)
         return self.add(PrimitiveOp([obj, key], py_getattr_op, line))
 
-    def py_call(self, function: Value, args: List[Value], line: int) -> Value:
-        arg_boxes = [self.box(arg) for arg in args]  # type: List[Value]
-        return self.add(PrimitiveOp([function] + arg_boxes, py_call_op, line))
-
-    def py_method_call(self, obj: Value, method: Value, args: List[Value], line: int) -> Value:
-        arg_boxes = [self.box(arg) for arg in args]  # type: List[Value]
-        return self.add(PrimitiveOp([obj, method] + arg_boxes, py_method_call_op, line))
+    def py_call(self,
+                function: Value,
+                arg_values: List[Value],
+                line: int,
+                arg_kinds: Optional[List[int]] = None,
+                arg_names: Optional[List[Optional[str]]] = None) -> Value:
+        """Call to non-mypyc-generated function, e.g. some python builtin functions."""
+        # Box all arguments since we are invoking a non-mypyc-generated function.
+
+        if (arg_kinds is None) or all(kind == ARG_POS for kind in arg_kinds):
+            arg_boxes = [self.box(value) for value in arg_values]
+            return self.add(PrimitiveOp([function] + arg_boxes, py_call_op, line))
```

Comment:
It now occurs to me that if we use the `primitive_op` method, we can avoid having to manually box arguments.

Reply:
Changed. Do we also want to refactor `self.add(PrimitiveOp...` elsewhere?